core/src/test/resources/org/apache/calcite/test/TopDownOptTest.xml

```diff
@@ -318,6 +318,31 @@ EnumerableProject(ENAME=[$0], JOB=[$1], EXPR$2=[$2], ENAME0=[$3], JOB0=[$4], SAL
     EnumerableSort(sort0=[$4], sort1=[$0], dir0=[ASC], dir1=[ASC])
       EnumerableProject(ENAME=[$0], JOB=[$1], SAL=[$2], COMM=[$3], JOB0=[CAST($1):BIGINT NOT NULL])
         EnumerableTableScan(table=[[CATALOG, SALES, BONUS]])
+]]>
+        </Resource>
+    </TestCase>
+    <TestCase name="testSortProjectDerive5">
+        <Resource name="sql">
+            <![CDATA[
+select empno*-1, ename, job from
+(select * from sales.emp order by empno, ename, job limit 10) order by ename, job
+]]>
+        </Resource>
+        <Resource name="planBefore">
+            <![CDATA[
+LogicalSort(sort0=[$1], sort1=[$2], dir0=[ASC], dir1=[ASC])
+  LogicalProject(EXPR$0=[*($0, -1)], ENAME=[$1], JOB=[$2])
+    LogicalSort(sort0=[$0], sort1=[$1], sort2=[$2], dir0=[ASC], dir1=[ASC], dir2=[ASC], fetch=[10])
+      LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8])
+        LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+        </Resource>
+        <Resource name="planAfter">
+            <![CDATA[
+EnumerableProject(EXPR$0=[*($0, -1)], ENAME=[$1], JOB=[$2])
+  EnumerableLimit(fetch=[10])
+    EnumerableSort(sort0=[$0], sort1=[$1], sort2=[$2], dir0=[ASC], dir1=[ASC], dir2=[ASC])
+      EnumerableTableScan(table=[[CATALOG, SALES, EMP]])
```

Comment:
The plan is wrong

Reply:
Why? 

The top project does not need to enforce a sort because inner sort has provide the needed ordering.

Reply:
I think a wrong plan is even though inner sort provide the required ordering, the top projection still enforces a sort (so not good).

Reply:
Bad plan is still correct plan. But this is incorrect plan.
The input is sorted by empno, ename, job .
We want the output sorted by ename, job.

empno | ename | job
------------ | ------------- | -------------
1 | 1 | 1
1 | 2 | 1
2 | 1 | 1
2 | 2 | 1